### PR TITLE
Fixed #32575 -- Added support for SpatiaLite 5.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

**Review by felixxm — COMMENTED** [review]
@claudep Thanks :+1: 

`ST_Transform()` on SpatiaLite is affected by [PROJ](https://www.gaia-gis.it/fossil/libspatialite/wiki?name=PROJ.6) so I fixed two tests with PROJ 7.X.

I'm not sure how to get rid of warnings related with `IsValid()` and `MakeValid()` calls :thinking: :
```bash
$ ./runtests.py gis_tests.geoapp
...
System check identified no issues (1 silenced).
.........s.........s.......s............ss..GEOS warning: Self-intersection at or near point 1 0
GEOS warning: Self-intersection at or near point 1 0
......s...s..GEOS warning: Self-intersection at or near point 1 0
RTTOPO warning: Self-intersection at or near point 1 0

RTTOPO warning: Hole lies outside shell at or near point 10 0

..........s.......s......s...........
----------------------------------------------------------------------
Ran 94 tests in 1.509s

```

**Comment by claudep** [comment]
> I'm not sure how to get rid of warnings related with `IsValid()` and `MakeValid()` calls thinking :

I'm not sure either, possibly impossible?

**Comment by felixxm** [comment]
@claudep Thanks again :+1: I pushed small edits and rebased on #14235.

> I'm not sure either, possibly impossible?

Yes, probably :confused:

File: tests/gis_tests/distapp/tests.py [hunk 1]
```diff
@@ -368,9 +368,15 @@ def test_distance_projected(self):
...
 
         # Original query done on PostGIS, have to adjust AlmostEqual tolerance
-        # for Oracle.
-        tol = 2 if connection.ops.oracle else 5
-
+        # for Oracle and SpatiaLite.
+        if connection.ops.spatialite:
+            # Different PROJ versions use different transformations, all are
+            # correct as having a 1 meter accuracy.
+            tol = -1
```
Comment (claudep) [inline tests/gis_tests/distapp/tests.py:375]: What about using the lower tolerance for all backends?

Comment (felixxm) [reply tests/gis_tests/distapp/tests.py:375]: Yes I was thinking about this. Personally, I don't think catching small accuracy errors is that important here so let's use `-1` for all DBs :+1:

**Pull request merged**